pandas/io/formats/format.py

```diff
@@ -1392,7 +1392,10 @@ def _format(x):
         fmt_values = []
         for i, v in enumerate(vals):
             if not is_float_type[i] and leading_space or self.formatter is not None:
-                fmt_values.append(f" {_format(v)}")
+                if leading_space:
+                    fmt_values.append(f" {_format(v)}")
+                else:
+                    fmt_values.append(f"{_format(v)}")
```

Comment:
```suggestion
```
It seems this if/else is already handled by the else block below. I think a smaller change is sufficient. WDYT? @reddyrg1 @mroeschke @phofl